src/main/java/org/assertj/core/internal/Uris.java

```diff
@@ -76,6 +77,12 @@ public void assertHasHost(AssertionInfo info, URI actual, String expected) {
     if (!Objects.equals(actual.getHost(), expected)) throw failures.failure(info, shouldHaveHost(actual, expected));
   }
 
+  public void assertHasNoHost(AssertionInfo info, URI actual) {
+    assertNotNull(info, actual);
+    if (actual.getHost() != null && !actual.getHost().isEmpty())
```

Comment:
The empty check has been added as it appears that the URL can return either-or. Seems that you can specify the host as null manually during initialisation but the constructor that parses those expressions will always use an emptystring.

Reply:
Amended tests to explicitly check both of these cases in addition to the parsing case.

Reply:
I did some tries and I'm actually getting the opposite result: when `host` is either `null` or `""` in the constructor, `getHost()` always returns `null`.